packages/react-notion-x/src/third-party/collection-view-calendar.tsx

```diff
@@ -0,0 +1,436 @@
+  'Jan',
+  'Feb',
+  'Mar',
+  'Apr',
+  'May',
+  'Jun',
+  'Jul',
+  'Aug',
+  'Sep',
+  'Oct',
+  'Nov',
+  'Dec'
+]
+let currentMonth = 0
+
+export const CollectionViewCalendar: React.FC<CollectionViewProps> = ({
+  collection,
+  collectionView,
+  collectionData
+}) => {
+  const isGroupedCollection = collectionView?.format?.collection_group_by
+
+  if (isGroupedCollection) {
+    const collectionGroups = getCollectionGroups(
+      collection,
+      collectionView,
+      collectionData
+    )
+
+    return collectionGroups.map((group, index) => (
+      <CollectionGroup
+        key={index}
+        {...group}
+        collectionViewComponent={Calendar}
+      />
+    ))
+  }
+
+  const blockIds =
+    (collectionData['collection_group_results']?.blockIds ??
+      collectionData['results:relation:uncategorized']?.blockIds ??
+      collectionData.blockIds) ||
+    defaultBlockIds
+
+  return (
+    <Calendar
+      collectionView={collectionView}
+      collection={collection}
+      blockIds={blockIds}
+    />
+  )
+}
+
+function Calendar({ blockIds, collectionView, collection }) {
+  const {
+    showCalendarControls,
+    startWeekOnMonday,
+    components,
+    recordMap,
+    mapPageUrl
+  } = useNotionContext()
+
+  const [weeksArr, setWeeksArr] = React.useState(
+    getWeeksInMonth(
+      currentYear.getFullYear(),
+      currentYear.getMonth(),
+      startWeekOnMonday
+    )
+  )
+
+  // Check if collection view is a standalone page or not
+  const isCollectionViewPage =
+    recordMap.block[Object.keys(recordMap.block)[0]].value.type ==
+    'collection_view_page'
+
+  const nextMonth = () => {
+    if (currentYear.getMonth() == 11) {
+      currentYear.setFullYear(currentYear.getFullYear() + 1)
+      currentYear.setMonth(0)
+    } else {
+      currentYear.setMonth(currentYear.getMonth() + 1)
+    }
+
+    currentMonth = 0
+
+    setWeeksArr(
+      getWeeksInMonth(
+        currentYear.getFullYear(),
+        currentYear.getMonth(),
+        startWeekOnMonday
+      )
+    )
+  }
+  const prevMonth = () => {
+    if (currentYear.getMonth() == 0) {
+      currentYear.setFullYear(currentYear.getFullYear() - 1)
+      currentYear.setMonth(11)
+    } else {
+      currentYear.setMonth(currentYear.getMonth() - 1)
+    }
+
+    currentMonth = 0
+
+    setWeeksArr(
+      getWeeksInMonth(
+        currentYear.getFullYear(),
+        currentYear.getMonth(),
+        startWeekOnMonday
+      )
+    )
+  }
+  const nowMonth = () => {
+    currentYear.setMonth(new Date().getMonth())
+    currentYear.setFullYear(new Date().getFullYear())
+
+    currentMonth = 0
+
+    setWeeksArr(
+      getWeeksInMonth(
+        currentYear.getFullYear(),
+        currentYear.getMonth(),
+        startWeekOnMonday
+      )
+    )
+  }
+
+  // Check on page load if the collection is a page or not
+  React.useEffect(() => {
+    if (isCollectionViewPage) {
+      document.querySelector('.notion-page').classList.add('notion-full-width')
```

Comment:
Hmmmmmmmmm this is definitely not where this logic belongs; it should be in `block.tsx` so we're not imperatively modifying the DOM, which is not guaranteed to remain consistent if React re-renders the top-level. This could happen, for instance, if the app switches some top-level context or swaps dark mode.